### ISSUE TITLE
[apex] ApexSharingViolations violation for classes without modifiers includes all lines as location of the violation

### ISSUE THREAD

**Issue by jag-j** [pmd/pmd#5511]
**Affects PMD Version:** 7.9.0

**Rule:** ApexSharingViolations
https://docs.pmd-code.org/pmd-doc-7.9.0/pmd_rules_apex_security.html#apexsharingviolations

**Description:**
ApexSharingViolations violations for classes without modifiers includes all lines. Since Apex expects modifiers for the main class in a file, this is typically only seen in inner classes as demonstrated in the code sample below.

**Code Sample demonstrating the issue:**

```java
public class ApexSharingViolations {
    public void violateSharing() {
        // Querying records without honoring sharing rules
        List<Account> accounts = [SELECT Id, Name FROM Account];
    }

    class ApexSharingViolationsInnerClass {
        public void violateSharing() {
            // Querying records without honoring sharing rules
            List<Account> accounts = [SELECT Id, Name FROM Account];
        }
    }
}
```

**Expected outcome:**

1. In the above class, violation is reported for both the classes ApexSharingViolations and ApexSharingViolationsInnerClass, as expected.
2. Violation for ApexSharingViolations has the class declaration line (i.e. public class ApexSharingViolations) as the Location information.
3. But for the violation for ApexSharingViolationsInnerClass has the entire class lines as Location Information as below:
      "locations": [
        {
          "file": "ApexSharingViolations.cls",
          "startLine": 7,
          "startColumn": 5,
          "endLine": 12,
          "endColumn": 6
        }
4. We expect this to be only reported on the class declaration line (line 7 in the above shared code).

**Running PMD through:** *CLI*
